go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -633,6 +635,26 @@ func (srv *mysqlServer) shutdownMysqlProtocolAndDrain() {
 		signal.Stop(srv.sigChan)
 	}
 
+	if mysqlDrainOnTerm {
+		// We wait for connected clients to drain, instead of just
+		// active (in transaction) clients
+		if srv.vtgateHandle == nil {
+			// server never started properly?
+			return
+		}
+
+		log.Infof("Starting drain loop, waiting for all clients to disconnect")
+		reported := time.Now()
+		for srv.vtgateHandle.numConnections() > 0 {
+			if time.Since(reported) > 5*time.Second {
+				log.Infof("Still waiting for client connections to drain (%d connected)...", srv.vtgateHandle.numConnections())
+				reported = time.Now()
+			}
+			time.Sleep(1000 * time.Millisecond)
+		}
+		return
+	}
```

Comment:
With Application side connection pooling this will most likely not change and VTGate can remain in the same state before being shut down.